cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/distributed/runtimejob/DefaultRuntimeJob.java

```diff
@@ -706,14 +738,50 @@ private InetSocketAddress getTrafficRelayTarget() {
         int port = GSON.fromJson(reader, ServiceSocksProxyInfo.class).getPort();
         return port == 0 ? null : new InetSocketAddress(InetAddress.getLoopbackAddress(), port);
       } catch (Exception e) {
-        OUTAGE_LOG.warn("Failed to open service proxy file {}", serviceProxyFile, e);
+        OUTAGE_LOG.warn("Failed to open service proxy file {}",
+            serviceProxyFile, e);
         return null;
       }
     }
 
     private File getServiceProxyFile() {
-      return new File("/tmp",
-          Constants.RuntimeMonitor.SERVICE_PROXY_FILE + "-" + programRunId.getRun() + ".json");
+      return new File("/tmp", Constants.RuntimeMonitor.SERVICE_PROXY_FILE + "-"
+                              + programRunId.getRun() + ".json");
+    }
+  }
+
+  private class ProgramStateListener extends AbstractListener {
```

Comment:
nit: can it be static or there are parent refs?

Reply:
Yes, it can be static. Change it to a static inner class.